cmd/registry/scoring/score_test.go

```diff
@@ -0,0 +1,648 @@
+	}{
+		{
+			desc: "spec pattern",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/-/specs/-",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+		},
+		{
+			desc: "specific api match",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/petstore/versions/-/specs/-",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+		},
+		{
+			desc: "specific api no match",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/test/versions/-/specs/-",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+			wantErr:      true,
+		},
+		{
+			desc: "specific version match",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/1.0.0/specs/-",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+		},
+		{
+			desc: "specific version no match",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/2.0.0/specs/-",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+			wantErr:      true,
+		},
+		{
+			desc: "specific spec match",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/-/specs/openapi.yaml",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+		},
+		{
+			desc: "specific spec no match",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/-/specs/swagger.yaml",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml",
+			wantErr:      true,
+		},
+		{
+			desc: "artifact pattern error",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/-/specs/-/artifacts/lint-spectral",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml/artifacts/lint-spectral",
+			wantErr:      true,
+		},
+		{
+			desc: "target and resource mismatch",
+			targetPattern: &rpc.ResourcePattern{
+				Pattern: "apis/-/versions/-/specs/-",
+			},
+			resourceName: "projects/pattern-test/locations/global/apis/petstore/versions/1.0.0",
+			wantErr:      true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			resourceName, _ := patterns.ParseResourcePattern(test.resourceName)
+			gotErr := matchResourceWithTarget(test.targetPattern, resourceName, "projects/pattern-test/locations/global")
+			if test.wantErr && gotErr == nil {
+				t.Errorf("expected matchResourceWithTarget() to return errors")
+			}
+
+			if !test.wantErr && gotErr != nil {
+				t.Errorf("matchResourceWithTarget() returned unexpected error: %s", gotErr)
+			}
+		})
+	}
+}
+
+func TestProcessScoreFormula(t *testing.T) {
+	tests := []struct {
```

Comment:
There's only one test in this table. Unless you have immediate plans to add tests here, consider removing the table and having the test case on its own in the function.